src/install/index.js

```diff
@@ -250,27 +271,22 @@ const deploySteps = (apps, mode, deployDoc, firstRun, ddoc) => {
     .then(getApplicationDdoc)
     .then(ddoc => {
       return stage('horti.stage.deploying', 'Deploying new installation')
-        .then(() => performDeploy(apps, mode, deployDoc, ddoc, firstRun))
-        .then(() => stage('horti.stage.postCleanup', 'Post-deploy cleanup, installation complete'))
-        .then(() => postCleanup(deployDoc));
+        .then(() => performDeploy(mode, deployDoc, ddoc, firstRun))
+        .then(() => {
+          stage('horti.stage.postCleanup', 'Post-deploy cleanup, installation complete');
```

Comment:
doesn't stage return a promise (because it writes to the staging doc)? It shouldn't execute at the same time as `postCleanup` (ie should be in its own then sequence)